src/Foundatio.RabbitMQ/Messaging/RabbitMQMessageBus.cs

```diff
@@ -117,21 +131,22 @@ public class RabbitMQMessageBus : MessageBusBase, IMessageBus {
                 return;
             _logger.Trace("Message Publish: {messageType}", messageType.FullName);
 
-            // RabbitMQ only supports delayed messages with a third party plugin.
-            if (delay.HasValue && delay.Value > TimeSpan.Zero) {
-                await AddDelayedMessageAsync(messageType, message, delay.Value).AnyContext();
-                return;
-            }
-
             var data = await _serializer.SerializeAsync(new MessageBusData {
                 Type = messageType.AssemblyQualifiedName,
                 Data = await _serializer.SerializeToStringAsync(message).AnyContext()
             }).AnyContext();
             
             var basicProperties = _publisherChannel.CreateBasicProperties();
             basicProperties.Persistent = _persistent;
-            basicProperties.Expiration = _defaultMessageTimeToLive.Milliseconds.ToString();
-
+            basicProperties.Expiration = _defaultMessageTimeToLive.TotalMilliseconds.ToString(CultureInfo.InvariantCulture);
+            // RabbitMQ only supports delayed messages with a third party plugin called "rabbitmq_delayed_message_exchange"
+            if (_delayedExchange && delay.HasValue && delay.Value > TimeSpan.Zero) {
```

Comment:
Yes, What do people normally do? You'd think that I'd always want a delayed exchange as it gives me both non delayed and delayed messaging support (right?). If we can remove an option and have the best of both worlds (either delayed messages via in process support or offloaded onto rabbitmq) that would be awesome :)